src/main/dialogs/promotionWindowDialog.js

```diff
@@ -0,0 +1,222 @@
+/**
+ * Promotion Window page BrowserWindow Dialog
+ *
+ * Introduces a component that uses an Electron BrowserWindow to represent a "Promotion Window" dialog.
+ *
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+
+var fluid = require("infusion");
+
+require("./basic/dialog.js");
+
+var gpii = fluid.registerNamespace("gpii");
+
+/**
+ * Component that represents the Promotion Window dialog
+ */
+fluid.defaults("gpii.app.promotionWindowDialog", {
+    gradeNames: ["gpii.app.dialog"],
+
+    siteConfig: {
+        promoContentUrl: null,
+        showPromotionWindow: null,
+        showCloseButton: null,
+        width: 200,
+        height: 200,
+        centered: false,
+        resizable: false,
+        movable: false,
+        skipTaskbar: false,
+        frame: false,
+        transparent: false,
+        alwaysOnTop: false,
+        closeDelay: 0,
+        showDelay: 0
+    },
+
+    closeDelay: "{that}.options.siteConfig.closeDelay",
+    showDelay: "{that}.options.siteConfig.showDelay",
+    // some delay, because of slow rearranging of windows taskbar
+    repositionDelay: 1900,
+
+    config: {
+        attrs: {
+            width: "{that}.options.siteConfig.width",
+            height: "{that}.options.siteConfig.height",
+            resizable: "{that}.options.siteConfig.resizable",
+            movable: "{that}.options.siteConfig.resizable.movable",
+            skipTaskbar: "{that}.options.siteConfig.resizable.skipTaskbar",
+            frame: "{that}.options.siteConfig.resizable.frame",
```

Comment:
These three appear to be at the wrong path - as seen in siteconfig they are at top-level paths such as {that}.options.siteConfig.frame